packages/neuron-ui/src/components/NervosDAO/hooks.ts

```diff
@@ -436,35 +437,75 @@ export const useUpdateWithdrawList = ({
   setWithdrawList: React.Dispatch<React.SetStateAction<Map<string, string | null>>>
 }) =>
   useEffect(() => {
-    Promise.all(
-      records.map(async ({ outPoint, depositOutPoint, blockHash }) => {
-        if (!tipBlockHash) {
-          return null
-        }
-        const withdrawBlockHash = depositOutPoint ? blockHash : tipBlockHash
-        const formattedDepositOutPoint = depositOutPoint
-          ? {
-              txHash: depositOutPoint.txHash,
-              index: `0x${BigInt(depositOutPoint.index).toString(16)}`,
-            }
-          : {
-              txHash: outPoint.txHash,
-              index: `0x${BigInt(outPoint.index).toString(16)}`,
-            }
-        return calculateDaoMaximumWithdraw(formattedDepositOutPoint, withdrawBlockHash).catch(() => null)
-      })
-    )
-      .then(res => {
-        const withdrawList = new Map()
-        if (tipBlockHash) {
-          records.forEach((record, idx) => {
-            const key = getRecordKey(record)
-            withdrawList.set(key, res[idx])
+    if (!tipBlockHash) {
+      setWithdrawList(new Map())
+      return
+    }
+    const depositOutPointHashes = records.map(v => v.depositOutPoint?.txHash ?? v.outPoint.txHash)
+    const txMap = new Map<CKBComponents.Hash, CKBComponents.TransactionWithStatus>()
+    ckbCore.rpc
+      .createBatchRequest<'getTransaction', string[], CKBComponents.TransactionWithStatus[]>(
+        depositOutPointHashes.map(v => ['getTransaction', v])
+      )
+      .exec()
+      .then(txs => {
+        const committedTx = txs.filter(v => v.txStatus.status === 'committed')
+        committedTx.forEach((tx, idx) => {
+          txMap.set(depositOutPointHashes[idx], tx)
```

Comment:
`depositOutPointHashes[idx]` and `tx` may be mismatched because `committedTx` is a subset of `depositOutPointHashes`(filtered by `'status === 'committed'`)

It should be `txMap.set(tx.transaction.hash, tx)`

Reply:
I fixed it by use `txs` to loop. And moving it nearby when it was used. https://github.com/nervosnetwork/neuron/pull/2530/commits/95d3bbe5a9d6bcaa507fa6fe5f651b9ddb06285b